src/Core/Request/Customers/CustomerLoginRequest.php

```diff
@@ -197,31 +202,31 @@ public function setAnonymousCart($anonymousCart)
     /**
      * @param string $email
      * @param string $password
-     * @param string $anonymousCartId
+     * @param CartReference|string $anonymousCart
      * @param Context $context
      * @return static
      */
-    public static function ofEmailAndPassword($email, $password, $anonymousCartId = null, Context $context = null)
+    public static function ofEmailAndPassword($email, $password, $anonymousCart = null, Context $context = null)
     {
-        return new static($email, $password, $anonymousCartId, $context);
+        return new static($email, $password, $anonymousCart, $context);
     }
 
     /**
      * @param string $email
      * @param string $password
      * @param bool $updateProductData
-     * @param string $anonymousCartId
+     * @param CartReference|null $anonymousCart
      * @param Context $context
      * @return static
      */
     public static function ofEmailPasswordAndUpdateProductData(
         $email,
         $password,
         $updateProductData,
-        $anonymousCartId = null,
+        CartReference $anonymousCart = null,
```

Comment:
Breaking change as you limit it now to CartReference only

Btw the setAnonymousCartId method should set the anonymousCart field to null as both fields are exclusive or connected

Reply:
I don't know why it didn't update this from my locale I put CartReference|string as type hinting